asdf/commands/edit.py

```diff
@@ -0,0 +1,529 @@
+    print("The edited text can then be saved to the ASDF file of your choice")
+    print("using 'asdftool edit -s -f <edited text file> -o <ASDF file>.")
+    print('-' * 70)
+    print("Note: This is meant to be a lightweight text editing tool of")
+    print("      ASDF .If the edited text is larger than the YAML portion")
+    print("      of the ASDF file to be written to, the edits may not be")
+    print("      able to saved.")
+    print(f"{delim}\n")
+
+    return
+
+def buffer_edited_text ( edited_text, orig_text ) :
+    """ There is more text in the original ASDF file than in the edited text,
+    so we will buffer the edited text with spaces.
+    """ 
+    diff = len(orig_text) - len(edited_text)
+    if diff<1 :
+        print("Error: shouldn't be here.")
+        sys.exit(1)
+
+    wdelim = b'\r\n...\r\n'
+    ldelim = b'\n...\n'
+    if edited_text[-len(wdelim):]==wdelim :
+        delim = wdelim
+    elif edited_text[-len(ldelim):]==ldelim :
+        delim = ldelim
+    else:
+        # Mabye raise exception
+        print("Unrecognized YAML delimiter ending the YAML text.")
+        print(f"It should be {wdelim} or {ldelim}, but the")
+        print(f"last {len(wdelim)} bytes are {edited_text[-len(wdelim):]}.")
+        sys.exit(1)
+
+    # May not be correct.  If on Windows use '\r\n'.
+    buffered_text = edited_text[:-len(delim)] + b'\n' + b' '*(diff-1) + delim
+    return buffered_text, diff-1 
+
+
+def add_buffer_to_new_text ( edited_text, buffer_size ) :
+    """ Adds buffer to edited text.
+    """
+    wdelim = b'\r\n...\r\n'
+    ldelim = b'\n...\n'
+    if edited_text[-len(wdelim):]==wdelim :
+        delim = wdelim
+    elif edited_text[-len(ldelim):]==ldelim :
+        delim = ldelim
+    else:
+        # Maybe raise exception
+        print("Unrecognized YAML delimiter ending the YAML text.")
+        print(f"It should be {wdelim} or {ldelim}, but the")
+        print(f"last {len(wdelim)} bytes are {edited_text[-len(wdelim):]}.")
+        sys.exit(1)
+
+    buf = b' ' * buffer_size
+    buffered_text = edited_text[:-len(delim)] + b'\n' + buf + delim
+
+    return buffered_text 
+
+def compute_block_index_blocks ( start, asdf_blocks ) :
+    """ Computes new block index and strips any data after last found block.
+    """
+    if constants.BLOCK_MAGIC!=asdf_blocks[:len(constants.BLOCK_MAGIC)] :
+        return [], asdf_blocks      # Not sure if this should happen
+
+    # Minimum block header is 
+    # 4 bytes of magic number
+    # 2 bytes of header length, after the length field (min 48)
+    # 4 bytes flag
+    # 4 bytes compression
+    # 8 bytes allocated size
+    # 8 bytes used (on disk) size
+    # 8 bytes data size
+    # 16 bytes checksum
+    bmlen = len(constants.BLOCK_MAGIC)
+    min_header = bmlen + 2 + 48  
+    uidx = 22
+    bindex = [start]
+    k = 0
+    while len(asdf_blocks) - k > min_header :
+        hsz = struct.unpack(">H",asdf_blocks[k+bmlen:k+bmlen+2])[0]
```

Comment:
There is a [header struct-like thing in the Block class](https://github.com/asdf-format/asdf/blob/d0a1c0a055233d7b91b05e8e8930cb5fe8f804c2/asdf/block.py#L782) that we can reuse here.